WordPress/Classes/Services/BlogJetpackSettingsService.swift

```diff
@@ -0,0 +1,210 @@
+
+    init(managedObjectContext context: NSManagedObjectContext) {
+        self.context = context
+    }
+
+    /// Sync ALL the Jetpack settings for a blog
+    ///
+    func syncJetpackSettingsForBlog(_ blog: Blog, success: @escaping () -> Void, failure: @escaping (Error?) -> Void) {
+
+        if !blog.supports(.jetpackSettings) {
+            success()
+            return
+        }
+
+        context.perform {
+            guard let blogInContext = self.context.object(with: blog.objectID) as? Blog,
+                let remote = BlogJetpackSettingsServiceRemote(wordPressComRestApi: blog.wordPressComRestApi()),
+                let blogDotComId = blogInContext.dotComID as? Int,
+                let blogSettings = blogInContext.settings else {
+                    success()
+                    return
+            }
+
+            var successfulFetches = 0
+            var fetchError: Error? = nil
+            var remoteJetpackSettings: RemoteBlogJetpackSettings? = nil
+            var remoteJetpackMonitorSettings: RemoteBlogJetpackMonitorSettings? = nil
+
+            // Create a dispatch group to wait for both calls.
+            let syncGroup = DispatchGroup()
+
+            syncGroup.enter()
+            remote.getJetpackSettingsForSite(blogDotComId,
+                                             success: { (remoteSettings) in
+                                                remoteJetpackSettings = remoteSettings
+                                                successfulFetches += 1
+                                                syncGroup.leave()
+                                             }, failure: { (error) in
+                                                fetchError = error
+                                                syncGroup.leave()
+                                             })
+
+            syncGroup.enter()
+            remote.getJetpackMonitorSettingsForSite(blogDotComId,
+                                                    success: { (remoteMonitorSettings) in
+                                                        remoteJetpackMonitorSettings = remoteMonitorSettings
+                                                        successfulFetches += 1
+                                                        syncGroup.leave()
+                                                    }, failure: { (error) in
+                                                        fetchError = error
+                                                        syncGroup.leave()
+                                                    })
+
+            syncGroup.notify(queue: DispatchQueue.main, execute: {
+                guard let remoteJetpackSettings = remoteJetpackSettings,
+                    let remoteJetpackMonitorSettings = remoteJetpackMonitorSettings else {
+                        failure(nil)
+                        return
+                }
+                if successfulFetches == 2 {
```

Comment:
Instead of a counter, maybe we can check if `remoteJetpackSettings` and `remoteJetpackMonitorSettings` are not nil

Reply:
Also, and specially considering that in the current implementation the VC will just log (but otherwise ignore) the error, why not call `.updateJetpackSettings`/`updateJetpackMonitorSettings` if any of them succeeds? Then we can call `failure` if there's a `fetchError` to pass.

Reply:
The guard above was already checking for that, I think I overcomplicated things here, will simplify.